packages/lib/event-types/utils/getSlugConflictsInChildren.ts

```diff
@@ -0,0 +1,22 @@
+import type { EventTypeSetupProps, FormValues } from "@calcom/features/eventtypes/lib/types";
+
+export const getSlugConflictsInChildren = ({
+  values,
+  eventType,
+}: {
+  values: FormValues;
+  eventType: EventTypeSetupProps["eventType"];
+}) => {
+  const conflicts = values.children.filter((child) => {
+    const isNewAssignment = !eventType.children.some((c) => c.owner.id === child.owner.id);
+    const isSlugPresentInChild = child.owner.eventTypeSlugs.includes(values.slug);
+    const isEventTypeSlugChanged = eventType.slug !== values.slug;
+
+    const isNewAssignmentWithConflict = isNewAssignment && isSlugPresentInChild;
+    const isExistingAssignmentWithConflict =
+      !isNewAssignment && isEventTypeSlugChanged && isSlugPresentInChild;
+    return isNewAssignmentWithConflict || isExistingAssignmentWithConflict;
+  });
```

Comment:
This works but why do we need to know if new users were assigned and conflicting or if parent event slug was changed and it is conflicting with existing users? Aren't we showing the same message.

Let's not complicate it and simply check if values.slug is present in children eventTypeSlugs.

Lmk if there is something I am missing here

Reply:
hey 
Thanks for the review. I have refactored the logic!!